src/coreclr/nativeaot/BuildIntegration/Microsoft.NETCore.Native.Unix.targets

```diff
@@ -151,6 +150,11 @@ The .NET Foundation licenses this file to you under the MIT license.
       <NativeLibrary Include="@(NetCoreAppNativeLibrary->'%(EscapedPath)')" />
     </ItemGroup>
 
+    <ItemGroup>
+      <!-- zlib must be added after System.IO.Compression.Native, order matters. -->
+      <NativeLibrary Condition="'$(UseSystemZlib)' != 'true'" Include="$(IlcSdkPath)libz.a" />
```

Comment:
Seems you moved this here because lines 145-150 need to happen first. We have the comment indicating that order matters and needs to be added after S.IO.Compression.Native. Should we have this comment improved by also mentioning this needs to be called _after escaping the NetCoreAppNativeLibrary items?

Reply:
I think the current comment is good enough. If you would like to see more verbose comment, please make github suggestion.

I think it is misleading to refer to the previous block as "escaping". The common meaning of escaping is to replace special characters with [escape sequences](https://en.wikipedia.org/wiki/Escape_sequence). It is not what it is doing.